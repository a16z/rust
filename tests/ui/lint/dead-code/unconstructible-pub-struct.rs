#![feature(never_type)]
#![deny(dead_code)]

pub struct T1(!);
pub struct T2(());
pub struct T3<X>(std::marker::PhantomData<X>);

pub struct T4 {
    _x: !,
}

pub struct T5<X> {
    _x: !,
    _y: X,
}

pub struct T6 {
    _x: (),
}

pub struct T7<X> {
    _x: (),
    _y: X,
}

pub struct T8<X> {
    _x: std::marker::PhantomData<X>,
}

pub struct T9<X> { //~ ERROR struct `T9` is never constructed
    _x: std::marker::PhantomData<X>,
    _y: i32,
}

fn main() {}
