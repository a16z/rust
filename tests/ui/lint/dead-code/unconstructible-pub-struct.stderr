error: struct `T9` is never constructed
  --> $DIR/unconstructible-pub-struct.rs:30:12
   |
LL | pub struct T9<X> {
   |            ^^
   |
note: the lint level is defined here
  --> $DIR/unconstructible-pub-struct.rs:2:9
   |
LL | #![deny(dead_code)]
   |         ^^^^^^^^^

error: aborting due to 1 previous error

